kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java

```diff
@@ -0,0 +1,448 @@
+     *
+     * @see org.kuali.kfs.fp.batch.service.ProcurementCardCreateDocumentService#autoApproveProcurementCardDocuments()
+     */
+    @Override
+    public boolean autoApproveProcurementCardDocuments() {
+    	//if auto approve is turned on
+		boolean autoApproveOn = super.parameterService.getParameterValueAsBoolean(
+				ProcurementCardAutoApproveDocumentsStep.class,
+				ProcurementCardDocumentRuleConstants.AUTO_APPROVE_DOCUMENTS_IND);
+
+        if (!autoApproveOn) { // no auto approve?  then skip out of here...
+            return true;
+        }
+
+        Collection<ProcurementCardDocument> pcardDocumentList = retrieveUAProcurementCardDocumentsToRoute(KewApiConstants.ROUTE_HEADER_ENROUTE_CD);
+
+        // get number of days and type for auto approve
+		int autoApproveNumberDays = Integer
+				.parseInt(super.parameterService.getParameterValueAsString(ProcurementCardAutoApproveDocumentsStep.class,
+						ProcurementCardDocumentRuleConstants.AUTO_APPROVE_NUMBER_OF_DAYS));
+
+        Timestamp currentDate = dateTimeService.getCurrentTimestamp();
+        for (ProcurementCardDocument pcardDocument: pcardDocumentList) {
+            try {
+
+                // prevent PCard documents from auto approving if they have capital asset info to collect
+                if(capitalAssetBuilderModuleService.hasCapitalAssetObjectSubType(pcardDocument)) {
+                    continue;
+                }
+
+                // if number of days in route is passed the allowed number, call doc service for super user approve
+                Timestamp docCreateDate = new Timestamp( pcardDocument.getDocumentHeader().getWorkflowDocument().getDateCreated().getMillis() );
+                if (KfsDateUtils.getDifferenceInDays(docCreateDate, currentDate) > autoApproveNumberDays) {
+                    // update document description to reflect the auto approval
+                    pcardDocument.getDocumentHeader().setDocumentDescription("Auto Approved On " + dateTimeService.toDateTimeString(currentDate) + ".");
+
+                    if ( LOG.isInfoEnabled() ) {
+                        LOG.info("Auto approving document # " + pcardDocument.getDocumentHeader().getDocumentNumber());
+                    }
+                    pcardDocument.setAutoApprovedIndicator(true);
+                    documentService.superUserApproveDocument(pcardDocument, "");
+                }
+            } catch (WorkflowException e) {
+                LOG.error("Error auto approving document # " + pcardDocument.getDocumentNumber() + " " + e.getMessage(),e);
+                throw new RuntimeException(e.getMessage(),e);
+            }
+        }
+
+        return true;
+    }
+    
+    /**
+     * Returns a list of all initiated but not yet routed procurement card documents, using the KualiWorkflowInfo service.
+     * @return a list of procurement card documents to route
+     */
+    //copy-pasted from super class to use correct object types (edu.arizona instead of org.kuali)
```

Comment:
Since this is directly introducing code duplication, I see it as necessary to explain every need for duplication (i.e. every duplicated method). The fact that all of these duplications have the same reason doesn't change the need to explain each individual duplication. While debugging an individual method, it may not be obvious or necessary to look at the commit or class as a whole; moving 'up' the explanation for the duplication would unnecessarily obfuscate the explanation for the duplication.